### PR TITLE
docs(hier-dep-inj): fix comp name, spelling in field name, copyedits

### PR DESCRIPTION
**Pull request by chalin**
- Move `@Input()` to be on setter rather than getter.
- Fix spelling of field `villaines` --> `villains`
- `TaxReturnComponent` --> `HeroTaxReturnComponent`
- Other misc copyedits

### CODE REVIEW

File: public/docs/ts/latest/guide/hierarchical-dependency-injection.jade [hunk 1]
```diff
@@ -93,9 +93,10 @@ figure.image-display
 
   Instead, provide the `VillainsService` in the `providers` metadata of the `VillainsListComponent` like this:
 
-+makeExample('hierarchical-dependency-injection/ts/src/app/villains-list.component.ts', 'metadata','src/app/villains-list.component.ts (metadata)')(format='.')
++makeExcerpt('src/app/villains-list.component.ts (metadata)')
+
 :marked
-  By providing `VillainsService` in the `VillainsListComponent` metadata &mdash; and nowhere else &mdash;, 
+  By providing `VillainsService` in the `VillainsListComponent` metadata &mdash; and nowhere else &mdash;
```
Comment (wardbell) [inline public/docs/ts/latest/guide/hierarchical-dependency-injection.jade]: The comma was correct but ugly.  Please rephrase:
>By providing `VillainsService` in the `VillainsListComponent` metadata and nowhere else,

File: public/docs/_examples/hierarchical-dependency-injection/ts/src/app/hero-tax-return.component.ts [hunk 2]
```diff
@@ -15,10 +15,11 @@ import { HeroTaxReturnService } from './hero-tax-return.service';
 export class HeroTaxReturnComponent {
   message = '';
   @Output() close = new EventEmitter<void>();
-  @Input()
+
   get taxReturn(): HeroTaxReturn {
     return this.heroTaxReturnService.taxReturn;
   }
+  @Input()
```
Comment (wardbell) [inline public/docs/_examples/hierarchical-dependency-injection/ts/src/app/hero-tax-return.component.ts]: Why did you move this here?  I realize it is technically true that the `@Input` applies to the setter but does it looks awful and it doesn't change Angular's behavior, right?

File: public/docs/ts/latest/guide/hierarchical-dependency-injection.jade (hunk 1 above)
Comment (chalin) [inline public/docs/ts/latest/guide/hierarchical-dependency-injection.jade]: done

File: public/docs/_examples/hierarchical-dependency-injection/ts/src/app/hero-tax-return.component.ts (hunk 2 above)
Comment (chalin) [inline public/docs/_examples/hierarchical-dependency-injection/ts/src/app/hero-tax-return.component.ts]: As a matter of style, we should encourage readers to associate `@Input` with setters. This way, documentation and refactoring tools, for example, will attribute the decorator to the proper class member.

Right, Angular's behavior isn't affected (though IMHO it might be incidental behavior that it happens to work when attached to the getter).

If you feel strongly about it, I can undo it. Let me know.

Comment (wardbell) [inline public/docs/_examples/hierarchical-dependency-injection/ts/src/app/hero-tax-return.component.ts]: I don't feel strongly about it  ... yet.

If we really believe that `@Input` should adorn the setter, we should find a place in the docs to say that and say why. If you have a moment to look for that place and propose language for it, I'd be grateful.

That would be a separate PR. I'm merging this one.
